src/main/java/net/mcreator/workspace/references/ReferencesFinder.java

```diff
@@ -0,0 +1,279 @@
+/*
+ * MCreator (https://mcreator.net/)
+ * Copyright (C) 2012-2020, Pylo
+ * Copyright (C) 2020-2023, Pylo, opensource contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package net.mcreator.workspace.references;
+
+import net.mcreator.blockly.data.BlocklyXML;
+import net.mcreator.element.GeneratableElement;
+import net.mcreator.element.parts.Sound;
+import net.mcreator.element.parts.procedure.Procedure;
+import net.mcreator.generator.mapping.MappableElement;
+import net.mcreator.minecraft.DataListEntry;
+import net.mcreator.ui.workspace.resources.TextureType;
+import net.mcreator.util.FilenameUtilsPatched;
+import net.mcreator.workspace.IWorkspaceProvider;
+import net.mcreator.workspace.Workspace;
+import net.mcreator.workspace.elements.ModElement;
+import net.mcreator.workspace.elements.SoundElement;
+import net.mcreator.workspace.resources.Model;
+import net.mcreator.workspace.resources.TexturedModel;
+
+import javax.annotation.Nullable;
+import java.io.File;
+import java.lang.reflect.AccessibleObject;
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.function.BiPredicate;
+import java.util.function.Predicate;
+
+public class ReferencesFinder {

```

Comment:
Here you could also check that fields are not primitive when doing recursion with "isPrimitive"

Reply:
Is that necessary? Wouldn't we just block access to such fields this way? Or you meant a specific line but comment got reset?

Reply:
Just that before you do recursive check on primitive, you stop before that happens for performance optimization

Reply:
Where exactly should I check for that? RF should already not continue recursion through anything different from lists, maps or custom objects that are all not primitives.

Reply:
> custom objects that are all not primitives

I did not check the code for that part yet, I just had this come to my mind and wrote it here just so I did not forget to check how this was done.

Btw how is the logic of custom objects implemented?

Reply:
You mean determining them or scanning them? RF considers an object to be custom if it's not in `java.base` module and not implementing `IWorkspaceProvider` (because otherwise checking it can overflow the stack). Then it just looks through all non-static fields and methods and scans deeper if the field/method filter executes for them.

Reply:
There are many other modules that are not java.base, though

Reply:
So is this part resolved?

Reply:
How was it resolved?

Reply:
Objects are only checked if they and their source fields/methods pass specified conditions and if these objects are in `ReferencesFinder`'s module and can't provide a workspace (unless they are of target type). Also,
>Here you could also check that fields are not primitive when doing recursion with "isPrimitive"

where should I look for this?

Reply:
https://docs.oracle.com/javase/8/docs/api/java/lang/Class.html#isPrimitive--

Reply:
Still, the method is only called on `GeneratableElement`s and objects that have passed `isCustomObject()` condition, in both cases primitive values are not encountered. Is this still needed as a sanity check?

Reply:
Probably not yes